nni/retiarii/strategy/base.py

```diff
@@ -6,10 +6,19 @@
 
 from ..graph import Model
 from ..mutator import Mutator
+from .utils import dry_run_for_formated_search_space
 
 
 class BaseStrategy(abc.ABC):
 
+    @classmethod
+    def get_model_space(self, base_model: Model, applied_mutators: List[Mutator]) -> None:
```

Comment:
This looks really weird. BaseStrategy is an interface, yet you put concrete implementations into it.